packages/material-ui-lab/src/MasonryItem/MasonryItem.js

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import { styled, useThemeProps } from '@material-ui/core/styles';
+import { getMasonryItemUtilityClass } from './masonryItemClasses';
+import MasonryContext from '../Masonry/MasonryContext';
+
+const useUtilityClasses = (styleProps) => {
+  const { classes } = styleProps;
+
+  const slots = {
+    root: ['root'],
+  };
+
+  return composeClasses(slots, getMasonryItemUtilityClass, classes);
+};
+
+const MasonryItemRoot = styled('div', {
+  name: 'MuiMasonryItem',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    return [styles.root];
+  },
+})(({ styleProps, theme }) => {
+  const gap = Number(theme.spacing(styleProps.spacing).replace('px', ''));
+  const rowSpan = styleProps.height ? Math.ceil(styleProps.height + gap) : 0;
+  return {
+    width: '100%',
+    [`& *`]: {
+      // all contents should have a width of 100%
+      objectFit: 'cover',
+      width: '100%',
+      height: '100%',
+    },
+    gridRowEnd: `span ${rowSpan}`,
+    paddingBottom: gap - 1,
+  };
+});
+
+const MasonryItem = React.forwardRef(function MasonryItem(inProps, ref) {
+  const props = useThemeProps({
+    props: inProps,
+    name: 'MuiMasonryItem',
+  });
+
+  const { spacing = 1 } = React.useContext(MasonryContext);
+  const { children, className, component = 'div', style, ...other } = props;
+  const styleProps = {
+    ...props,
+    spacing,
+  };
+
+  const classes = useUtilityClasses(styleProps);
+
+  return (
+    <MasonryItemRoot
+      as={component}
+      className={clsx(classes.root, className)}
+      ref={ref}
+      style={style}
```

Comment:
We can remove this if we are not handling the `style` prop. It will anyway be spread as part of other.